### PR TITLE
[BIT]回测问题修复：cast、Tensor.cast、var、nn.functional.gumbel_softmax

### PR DESCRIPTION
**Pull request by yuwu46**
### 修复情况

本次修复了 `paddle.cast`、`paddle.Tensor.cast`、`paddle.var`、`paddle.nn.functional.gumbel_softmax` 共 4 个 API 的回测精度问题，具体情况如下：
| PaddlePaddle API 名称 | 精度问题描述 | 解决方法 |
|:----------------------|:-------------|:---------|
| paddle.cast | 相关配置中含有字符串 “float”，需要转为 Python 内建的 float64 类型，原来修复[#pr130](https://github.com/PFCCLab/PaddleAPITest/pull/130)修复有误，在 base.py 中错误设置为 float32 | 在 base 文件 `convert_dtype_to_torch_type` 函数中，将 “float” 调整到 `return torch.float64`，目前仅 cast 和 Tensor.cast 会用到 “float” 和 “double”，不会影响其他 API 配置 |
| paddle.Tensor.cast | 与 paddle.cast 问题相同 | 与 paddle.cast 修复方式相同 |
| paddle.var | 当输入 Tensor 为标量时，两者输出表现不一致 | 添加 rule：`VarRule`，对 `x.dim() == 0` 的情况做特殊处理 |
| paddle.nn.functional.gumbel_softmax | 查看文档，该 API 会使用 gumbel 噪声，计算过程具有随机性 | 将 API 的相关配置移入 `random_calculation.txt` |

### 合并测试结果
将原来已通过的配置和 pending 配置合并后，共121380条，均能通过。

### CODE REVIEW

**Review by wanghuancoder — APPROVED** [review]
LGTM
